lib/ansible/plugins/action/assert.py

```diff
@@ -38,10 +38,13 @@ def run(self, tmp=None, task_vars=None):
         if 'that' not in self._task.args:
             raise AnsibleError('conditional required in "that" string')
 
-        msg = None
+        fail_msg = None
         success_msg = None
-        if 'msg' in self._task.args and isinstance(self._task.args['msg'], string_types):
-            msg = self._task.args['msg']
+        if 'fail_msg' in self._task.args and isinstance(self._task.args['fail_msg'], string_types):
+            fail_msg = self._task.args['fail_msg']
+        elif 'msg' in self._task.args and isinstance(self._task.args['msg'], string_types):
+            fail_msg = self._task.args['msg']
```

Comment:
something like this is probably simpler and more correct as it doesn't silently ignore a bad option type
```
fail_msg = self._task.args.get('fail_msg', self._task.args.get('msg'))
if not isinstance(fail_msg, string_types):
    raise AnsibleError('Incorrect type for fail message, expected string and got %s' % type(fail_msg)```

Reply:
@bcoca added more explanation in docstring and added code to check the type of fail_msg.